pandas/core/categorical.py

```diff
@@ -302,11 +306,31 @@ def copy(self):
         return Categorical(values=self._codes.copy(),categories=self.categories,
                            name=self.name, ordered=self.ordered, fastpath=True)
 
+    def astype(self, dtype):
+        """ coerce this type to another dtype """
+        if is_categorical_dtype(dtype):
+            return self
+        elif is_object_dtype(dtype):
+            return np.array(self)
```

Comment:
I don't like this, particularly because the dtype of the returned array will usually *not* be object.

Why don't you simply do `return np.array(self, dtype=dtype)` and allow any valid numpy dtype?